### PR TITLE
feat/FOROME-1310: add new info card of dataset

### PR DESCRIPTION
**Pull request by QSdmitrioul**
[TASK](https://quantori.atlassian.net/browse/FOROME-1310)

### CODE REVIEW

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export const INFO: ReadonlyArray<Row<keyof IDsInfo>> = [
+  { field: 'name', fieldName: 'Name' },
+  { field: 'kind', fieldName: 'Kind' },
+  { field: 'total', fieldName: 'Variants' },
+  {
+    // @ts-ignore
+    field: 'createTime',
+    fieldName: 'Created at',
+    render: (it: string) => new Date(it).toISOString(),
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts]: what means `it`? Don't use contractions

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+import cn from 'classnames'
+
+import { CardTitle } from '@ui/card-main'
+import { IDsInfoReceipt } from '@service-providers/dataset-level'
+import { Row } from './dataset-info.interfaces'
+
+export const renderRow =
+  <T,>(info: T) =>
+  (it: Row<keyof T>) => {
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx]: here too

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        <td>{it.render ? it.render(value) : value}</td>
+      </tr>
+    )
+  }
+
+export const renderAncestor = (it: IDsInfoReceipt) => {
+  const isFilter = it.kind === 'filter'
+
+  const title = `${
+    isFilter ? 'Filter applied in' : 'Decision tree code applied in'
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx]: I suppose copy text should be moved to en.ts

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.module.css [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        font-weight: bold;
+      }
+    }
+
+    &__conditions {
+      @apply typography-s;
+
+      & > span {
+        margin-top: 8px;
+        font-family: "Droid Sans Mono", monospace, "Droid Sans Fallback";
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.module.css]: "Droid Sans Mono", "Droid Sans Fallback", monospace

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import styles from './dataset-info.module.css'
+
+import { FC, useMemo } from 'react'
+import cn, { Argument } from 'classnames'
+import _ from 'lodash'
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: import startCase from 'lodash/startCase'

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  const isLoading = datasetStore.isLoading
+  const info = datasetStore.dsInfoData
+  const versions = info?.meta.versions
+  const receipts = info?.receipts
+
+  const versionsRaws = useMemo(
+    () =>
+      Object.keys(versions || {}).map(
+        (it: string): Row<keyof Versions> => ({
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: get rid of `it` everywhere, use something meaningful

File: src/core/hooks/use-fullscreen-view.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import { useCallback, useEffect, useState } from 'react'
+
+export const useFullScreenView = (
+  id: string,
+  callback?: (v: boolean) => void,
+): [boolean, () => void] => {
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  const toggleStoreFullScreen = useCallback(() => {
+    const element = document.querySelector(`#${id}`)
```
Comment (AndreyUstyumenko) [inline src/core/hooks/use-fullscreen-view.ts]: document.getElementById(id)

Comment (AndreyUstyumenko) [reply src/core/hooks/use-fullscreen-view.ts]: better to work with refs

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+import { INFO } from './dataset-info.constants'
+import { Row } from './dataset-info.interfaces'
+import { renderAncestor, renderRow } from './dataset-info.utils'
+
+interface IDatasetInfoProps {
+  className?: Argument
+}
+
+export const DatasetInfo: FC<IDatasetInfoProps> = observer(({ className }) => {
+  const id = 'InfoDetailsContaoner'
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: typo

File: src/service-providers/dataset-level/dataset-level.interface.ts [hunk 9]
```diff
@@ -52,8 +53,23 @@ export interface IDsInfoMeta {
...
+  base: string
+  conditions: Array<string | string[]>
+  'eval-update-info'?: [string | null, string | null]
+  'f-presentation'?: string[]
+  'p-presentation'?: [string, number, boolean | null][]
+  'filter-name'?: string
+  'dtree-name'?: string
+  'dtree-code'?: string
+  'panels-supply'?: { [key: string]: string[] }
+  kind: 'filter' | 'dtree'
```
Comment (AndreyUstyumenko) [inline src/service-providers/dataset-level/dataset-level.interface.ts]: required - above, optional - below

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+  const [isExpanded, toggle] = useFullScreenView(id)
+
+  const isLoading = datasetStore.isLoading
+  const info = datasetStore.dsInfoData
```
Comment (Lapk1n) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: const { isLoading, dsInfoData: info } = datasetStore

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  const versions = info?.meta.versions
+  const receipts = info?.receipts
+
+  const versionsRaws = useMemo(
```
Comment (Lapk1n) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: Rows

File: src/ui/card-main/card.tsx [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+import { FC } from 'react'
+import cn, { Argument } from 'classnames'
+
+interface ICardProps {
+  id?: string
+  className?: Argument
+}
+
+export const Card: FC<ICardProps> = ({ className, children, ...props }) => {
+  return (
```
Comment (Lapk1n) [inline src/ui/card-main/card.tsx]: no need for return

File: src/ui/card-main/card-title.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export const CardTitle: FC<ICardTitleProps> = ({
+  showExpandButton,
+  onExpand,
+  isExpanded,
+  children,
+  className,
+}) => {
+  return (
```
Comment (Lapk1n) [inline src/ui/card-main/card-title.tsx]: no need for return

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts (hunk 1 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx (hunk 2 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.module.css (hunk 4 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.module.css]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx (hunk 5 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx (hunk 6 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx (hunk 3 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx]: Add localization everywhere

File: src/service-providers/dataset-level/dataset-level.interface.ts (hunk 9 above)
Comment (QSdmitrioul) [reply src/service-providers/dataset-level/dataset-level.interface.ts]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx (hunk 11 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: Fixed

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx (hunk 10 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: Fixed

File: src/ui/card-main/card-title.tsx (hunk 13 above)
Comment (QSdmitrioul) [reply src/ui/card-main/card-title.tsx]: fixed

File: src/core/hooks/use-fullscreen-view.ts (hunk 7 above)
Comment (QSdmitrioul) [reply src/core/hooks/use-fullscreen-view.ts]: Use ref

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx (hunk 8 above)
Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.tsx]: Removed

**Comment by Lapk1n** [comment]
Screenshot 2022-07-19 at 19 22 10
The title font of the wizard cards is increased

**Comment by Lapk1n** [comment]
Screenshot 2022-07-19 at 19 24 35
And we lost the full width of the selected list item

**Comment by Lapk1n** [comment]
Screenshot 2022-07-19 at 19 28 27
None of the wizard scenarios cant fit without a scroll

**Comment by AndreyUstyumenko** [comment]
Line separators color does not match to one on the mockup
Screen Shot 2022-07-20 at 00 28 01

**Comment by AndreyUstyumenko** [comment]
'Info' menu item is not selected by default
Screen Shot 2022-07-20 at 00 31 11

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+  {
+    field: 'ancestors',
+    fieldName: t('home.infoPanel.info.rootDataset'),
+    optional: true,
+    render: (ancestors: string[][]) => {
+      if (ancestors.length === 0) {
+        return null
+      }
+
+      return ancestors[ancestors.length - 1][0]
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts:48]: Every dataset may have an infinite number of children, so this thing must be recursive until you find the very first ancestor.
UPD: Not a valid comment

Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.constants.ts:48]: Last ancestor is a root ancestor, what the problem, we don't need recursion here

**Comment by QSdmitrioul** [comment]
All comments was fixed

File: src/i18n/locales/en.ts [hunk 15]
```diff
@@ -107,6 +107,27 @@ export const en = {
...
+        filterName: 'Filter name: ',
+        dtreeName: 'Tree name: ',
+        updated: 'updated at {at} from {from}',
+        suppliedPanels: 'Supplied panels for type ',
+      },
+      info: {
+        name: 'Name',
+        kind: 'Kind',
+        variants: 'Variants',
+        cratedAt: 'Created At',
```
Comment (EvgeniySobolev) [inline src/i18n/locales/en.ts]: typo

**Review by EvgeniySobolev — APPROVED** [review]

File: src/i18n/locales/en.ts (hunk 15 above)
Comment (QSdmitrioul) [reply src/i18n/locales/en.ts]: Fixed

**Comment by AndreyUstyumenko** [comment]
Could you pay more attention to the Figma layout? They don't seem to match.
1) Something wrong with border radius. It exists only on two corners.
2) Each card on the layout fills 50% of the container
3) Table separators should have 1px of thickness
4) Look at the padding at the right side
5) Info card on the layout has a border
6) Card titles should be 16px by default
7) Shadows on Figma layout are obviously darker. You used the same properties but it's a subject to discuss at least with Anastasia. Shadows are rendered differently in editors and in browsers.
UPD: I reached out on Anastasia and asked her to pick better rgba properties for shadows in the browser.
Screen Shot 2022-07-21 at 01 41 57
Screen Shot 2022-07-21 at 01 42 28

**Comment by AndreyUstyumenko** [comment]
Moving the panel down at 1279px makes it unreadable on smaller screens. Better not to do anything, since we have a collapsing sidebar. As an option we can collapse the sidebar at 1279, but it's for another task.
Screen Shot 2022-07-21 at 02 33 40

File: src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx [hunk 16]
```diff
@@ -0,0 +1,88 @@
+import styles from './dataset-info.module.css'
```
Comment (AndreyUstyumenko) [inline src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx]: Utils are utility functions, these are React components. Please split them onto separate files and move them to /components folder.

Comment (QSdmitrioul) [reply src/pages/main/components/selected-dataset/components/dataset-info/dataset-info.utils.tsx]: Fixed

**Comment by Lapk1n** [comment]
Screenshot 2022-07-21 at 12 36 02
I think the last row shouldn't have border-bottom

**Comment by Lapk1n** [comment]
Screenshot 2022-07-21 at 12 32 20
Expand button doesn't work.

**Comment by Lapk1n** [comment]
Screenshot 2022-07-21 at 12 31 57
Wizard header with dataset name got lower font size

**Comment by QSdmitrioul** [comment]
> 7\. Shadows on Fi

According to Figma it should be. And also without it, table wouldn't look consistent

**Comment by QSdmitrioul** [comment]
All comments were fixed or resolved, except **card border and card shadow**, this comments wait answer from designer.

**Review by Lapk1n — APPROVED** [review]

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**